cmd/util/ledger/migrations/cadence_values_migration.go

```diff
@@ -0,0 +1,298 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/reporters"
+	"github.com/onflow/flow-go/cmd/util/ledger/util"
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/fvm/tracing"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/onflow/cadence/migrations"
+	"github.com/onflow/cadence/migrations/account_type"
+	"github.com/onflow/cadence/migrations/capcons"
+	"github.com/onflow/cadence/migrations/entitlements"
+	"github.com/onflow/cadence/migrations/string_normalization"
+	"github.com/onflow/cadence/migrations/type_value"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+)
+
+type CadenceBaseMigrator struct {
+	log             zerolog.Logger
+	reporter        reporters.ReportWriter
+	valueMigrations func(
+		inter *interpreter.Interpreter,
+		accounts environment.Accounts,
+		reporter *cadenceValueMigrationReporter,
+	) []migrations.ValueMigration
+}
+
+var _ AccountBasedMigration = (*CadenceBaseMigrator)(nil)
+var _ io.Closer = (*CadenceBaseMigrator)(nil)
+
+func (m *CadenceBaseMigrator) Close() error {
+	// Close the report writer so it flushes to file.
+	m.reporter.Close()
+	return nil
+}
+
+func (m *CadenceBaseMigrator) InitMigration(
+	log zerolog.Logger,
+	_ []*ledger.Payload,
+	_ int,
+) error {
+	m.log = log.With().Str("migration", "cadence-value-migration").Logger()
```

Comment:
Maybe the migration name should be a field of `CadenceBaseMigrator`, given that there will be multiple `CadenceBaseMigrator` instances (https://github.com/onflow/flow-go/pull/5192/files#diff-c0a5469b0a82ca259d28b12f20ff41aeeee0f9e2b98b2ed3117075d98ba6ea3bR143-R144, 
and https://github.com/onflow/flow-go/pull/5192/files#diff-c0a5469b0a82ca259d28b12f20ff41aeeee0f9e2b98b2ed3117075d98ba6ea3bR171-R172)
```suggestion
	m.log = log.With().Str("migration", m.name).Logger()
```